File: demo/python/udp_pair.py
from __future__ import annotations

import time

from unitransmit import UniTransmit


def main() -> None:
    u1 = UniTransmit("udp://0.0.0.0:9000?remote=127.0.0.1:9001")
    u2 = UniTransmit("udp://0.0.0.0:9001?remote=127.0.0.1:9000")

    def on_recv(data: bytes, ctx: dict) -> None:
        print(f"recv: {data!r} from {ctx}")

    u2.set_receive_callback(on_recv)

    print("send ->", u1.write(b"hello"))
    time.sleep(0.1)

    u1.close()
    u2.close()


if __name__ == "__main__":
    main()

File: demo/cpp/udp_pair.cpp
#include "cc/neolux/utils/unitransmit/Unitransmit.h"

#include <chrono>
#include <cstdint>
#include <iostream>
#include <string>
#include <thread>
#include <vector>

using cc::neolux::utils::unitransmit::UniTransmit;

int main() {
    UniTransmit a("udp://0.0.0.0:9000?remote=127.0.0.1:9001");
    UniTransmit b("udp://0.0.0.0:9001?remote=127.0.0.1:9000");

    a.write(std::string("hello"));
    std::this_thread::sleep_for(std::chrono::milliseconds(50));

    auto data = b.read_all();
    std::string out(reinterpret_cast<const char *>(data.data()), data.size());
    std::cout << "recv: " << out << "\n";

    return 0;
}
